Add index on publicKeyCredentialId in Doctrine mapping

Spomky opened the pull request:
This commit introduces a database index on the `publicKeyCredentialId` field in the `PublicKeyCredentialSource` mapping. The added index aims to optimize query performance for operations involving this column.

Target branch: 5.1.x
Resolves issue #665 

<!-- replace space with "x" in square brackets: [x] -->
- [ ] It is a Bug fix
- [x] It is a New feature
- [ ] Breaks BC
- [ ] Includes Deprecations

<!--
Fill in this template according to the PR you're about to submit.
Replace this comment by a description of what your PR is solving.

Please consider the following requirement:
* Modification of existing tests should be avoided unless deemed necessary.
* You MUST never open a PR related to a security issue. Contact Spomky in private at https://gitter.im/Spomky/
-->

ruudk commented on src/symfony/src/Resources/config/doctrine-mapping/PublicKeyCredentialSource.orm.xml:
```diff
@@ -5,6 +5,9 @@
     <mapped-superclass name="Webauthn\PublicKeyCredentialSource">
+        <indexes>
+            <index name="pkcid_idx" columns="publicKeyCredentialId"/>
+        </indexes>
```
What about making it a unique column instead? That already gives it an index, plus it guarantees it's unique.

Spomky committed:
Set publicKeyCredentialId as unique in ORM mapping

Marked the publicKeyCredentialId field as unique in the PublicKeyCredentialSource mapping configuration. This ensures database-level enforcement of uniqueness for this field, preventing duplicate entries.

Spomky committed:
Set publicKeyCredentialId as unique in ORM mapping

Marked the publicKeyCredentialId field as unique in the PublicKeyCredentialSource mapping configuration. This ensures database-level enforcement of uniqueness for this field, preventing duplicate entries.

nicodemuz commented:
I just deployed this to production via `php bin/console doctrine:schema:update --force --dump-sql`, and got the following error:

```
CREATE UNIQUE INDEX UNIQ_6AD0B4DF72A8BD77 ON public_key_credential_sources (public_key_credential_id);

 Updating database schema...


In ExceptionConverter.php line 118:
                                                                                                                                                                                             
  An exception occurred while executing a query: SQLSTATE[42000]: Syntax error or access violation: 1170 BLOB/TEXT column 'public_key_credential_id' used in key specification without a key length
```

I'm using MySQL.

Spomky commented:
Hello @nicodemuz,

This should be fixed in 5.1.1

nicodemuz commented:
> Hello @nicodemuz,
> 
> This should be fixed in 5.1.1

@Spomky The MySQL query becomes now:
```
ALTER TABLE public_key_credential_sources CHANGE public_key_credential_id public_key_credential_id TINYTEXT NOT NULL;
```

With an error:
```
An exception occurred while executing a query: SQLSTATE[42000]: Syntax error or access violation: 1170 BLOB/TEXT column 'public_key_credential_id' used in key specification without a key length
```

I just tested that it (doctrine:schema:update) works with:
```
<field name="publicKeyCredentialId" type="string" unique="true" length="250"/>
```

But I'm not sure if this will break anything.

fritzmg commented:
I can confirm the issue. The key length is always required when using keys (e.g. for a unique index) on fields like `text`, `tinytext`, `longtext` etc.
